### PR TITLE
Support buy ETH in the driver

### PR DESCRIPTION
**Pull request by cowfee**
Fixes https://github.com/cowprotocol/services/issues/1311.

Allow orders which buy ETH. Pass such orders to the solver wrapped in WETH. Properly interpret the price of WETH in the solution to be equal to the price of ETH.

I'm not happy with the code. A lot of edge cases appear due to the special handling of ETH and I haven't managed to come up with a good way to make them natural. I don't have any better ideas at the moment.

### Test Plan

Added an automated test.

### CODE REVIEW

**Review by MartinquaXD — APPROVED** [review]
I **think** everything checks out here.

File: crates/driver/src/boundary/settlement.rs [hunk 1]
```diff
@@ -393,3 +398,11 @@ fn to_big_decimal(value: bigdecimal::BigDecimal) -> num::BigRational {
     let numerator = num::BigRational::new(base, 1.into());
     numerator / ten.pow(exp.try_into().expect("should not overflow"))
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("boundary error: {0:?}")]
+    Boundary(anyhow::Error),
+    #[error("solution error: {0:?}")]
+    Solution(#[from] solution::Error),
```
[the remark this answers is not in the document]
Comment (nlordell) [reply crates/driver/src/boundary/settlement.rs:407]: Doesn't this error only matter internally though? Personally, I think its OK to leave it as "boundary error", as the internal error description should have enough information to identify what is going on.

[the remark this answers is not in the document]
Comment (nlordell) [reply crates/driver/src/boundary/settlement.rs:407]: Merging with this error message for now, we can come back and adjust this later if you disagree.

File: crates/driver/src/domain/competition/solution/mod.rs [hunk 2]
```diff
@@ -246,6 +250,60 @@ impl Solution {
...
+        // WETH instead of ETH. Once the driver receives the solution which fulfills an
+        // ETH order, a clearing price for ETH needs to be added, equal to the
+        // WETH clearing price.
+        if self.user_trades().any(|trade| trade.order.buys_eth()) {
+            // If no order buys WETH, the WETH price is not necessary, only the ETH price is
+            // needed. Remove the unneeded WETH price, which slightly reduces gas used by
+            // the settlement.
+            let mut prices = if self
+                .user_trades()
+                .all(|trade| trade.order.buy.token != self.weth.0)
```
Comment (nlordell) [inline crates/driver/src/domain/competition/solution/mod.rs:275]: No order can buy or sell WETH. You can have a settlement with:
- Order 1: sell WETH buy DAI
- Order 2: sell DAI  buy ETH

Which would require all three prices but get filtered out here.

**Review by nlordell — APPROVED** [review]

File: crates/driver/src/domain/competition/solution/mod.rs (hunk 2 above)
Comment (nlordell) [reply crates/driver/src/domain/competition/solution/mod.rs]: Fixed in 611791e6fb0c1800878ed1735830699c6b1e77a9
